src/main/java/io/jenkins/plugins/checks/status/FlowExecutionAnalyzer.java

```diff
@@ -0,0 +1,199 @@
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .collect(Collectors.toList());
+
+                Collections.reverse(location);
+
+                location.add(flowNode.getDisplayName());
+
+                nodeSummaryBuilder.append(String.format("### `%s`%n", String.join(" / ", location)));
+
+                nodeSummaryBuilder.append(String.format("%s in `%s` step", errorAction == null ? "Warning" : "Error", flowNode.getDisplayFunctionName()));
+                String arguments = ArgumentsAction.getStepArgumentsAsString(flowNode);
+                if (arguments == null) {
+                    nodeSummaryBuilder.append(".\n");
+                }
+                else {
+                    nodeSummaryBuilder.append(String.format(", with arguments `%s`.%n", arguments));
+                }
+
+                nodeTextBuilder.append(String.join("", Collections.nCopies(indentationStack.size() + 1, "  ")));
+                if (warningAction == null) {
+                    nodeTextBuilder.append(String.format("**Error**: *%s*", errorAction.getDisplayName()));
+                    String log = getLog(flowNode);
+                    if (StringUtils.isNotBlank(log)) {
+                        nodeSummaryBuilder.append(String.format("```%n%s%n```%n<details>%n<summary>Build log</summary>%n```%n%s```%n</details>%n",
+                                errorAction.getDisplayName(),
+                                log));
+                    }
+                }
+                else {
+                    nodeTextBuilder.append(String.format("**Unstable**: *%s*", warningAction.getMessage()));
+                    nodeSummaryBuilder.append(String.format("```%n%s%n```%n%n", warningAction.getMessage()));
+                }
+            }
+
+            nodeTextBuilder.append("\n");
+
+            summaryBuilder.append(nodeSummaryBuilder);
+            textBuilder.append(nodeTextBuilder);
+        });
+
+        return new ChecksOutput.ChecksOutputBuilder()
+                .withTitle(extractOutputTitle())
+                .withSummary(summaryBuilder.toString())
+                .withText(textBuilder.toString())
+                .build();
+    }
+
+    @CheckForNull
+    @SuppressFBWarnings("CRLF_INJECTION_LOGS")
+    private static String getLog(final FlowNode flowNode) {
+        LogAction logAction = flowNode.getAction(LogAction.class);
+        if (logAction == null) {
+            return null;
+        }
+        try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+            if (logAction.getLogText().writeLogTo(0, out) == 0) {
+                return null;
+            }
+            return out.toString(StandardCharsets.UTF_8.toString());
+        }
+        catch (IOException e) {
+            LOGGER.log(Level.WARNING, String.format("Failed to extract logs for step '%s'", flowNode.getDisplayName()), e);
```

Comment:
this should fix the findsec spotbugs log issue, technically it is a real issue some step could have a crazy display name, not very likely but...
```suggestion
            LOGGER.log(Level.WARNING, String.format("Failed to extract logs for step '%s'", flowNode.getDisplayName()).replaceAll("[\r\n]", ""), e);
```